src/nagini_translation/lib/viper_ast.py

```diff
@@ -510,6 +510,7 @@ def to_position(self, expr, vias, error_string: str=None,
     def is_heap_dependent(self, expr) -> bool:
         """
         Checks if the given expression contains an access to a heap location.
+        Does NOT check for calls to heap-dependent functions.
```

Comment:
That means it is not a recursive check, right?

Reply:
It is recursive in the sense that it checks the inner expressions if expr is a nested expression, but it doesn't follow the dependencies if you will; it just checks if there is a field access or something like that somewhere in the expression.